### PR TITLE
Add PrometheusSource webhook

### PR DESCRIPTION
**Pull request by MIBc**
Fixes: #1111

## Proposed Changes
- Add validation and default webhook for prometheus source.

### CODE REVIEW

File: prometheus/cmd/webhook/main.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		true,
+
+		// Extra validating callbacks to be applied to resources.
+		callbacks,
+	)
+}
+
+func main() {
+	ctx := webhook.WithOptions(signals.NewContext(), webhook.Options{
+		ServiceName: "prometheus-source-webhook",
```
Comment (lberk) [inline prometheus/cmd/webhook/main.go:95]: we use `logconfig.WebhookName()` below, could we please use it here too to stay consistent?

File: prometheus/config/500-webhook-config.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 The Knative Authors
```
Comment (lberk) [inline prometheus/config/500-webhook-config.yaml:1]: 2019?

File: prometheus/cmd/webhook/main.go (hunk 1 above)
Comment (MIBc) [reply prometheus/cmd/webhook/main.go:95]: Yes

Comment (MIBc) [reply prometheus/cmd/webhook/main.go]: done

File: prometheus/config/500-webhook-config.yaml (hunk 2 above)
Comment (MIBc) [reply prometheus/config/500-webhook-config.yaml]: done

**Pull request merged**